### PR TITLE
ipvs: add graceful termination unit tests

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
As per comment https://github.com/kubernetes/kubernetes/pull/77802#issuecomment-493643865, adding initial set of unit tests for IPVS's graceful termination logic. There's some coverage that's missing but wanted to keep the PR smaller for the first pass. 

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/util/ipvs/testing/fake.go [hunk 1]
```diff
@@ -27,47 +27,47 @@ import (
...
 type FakeIPVS struct {
 	Scheduler    string
-	Services     map[serviceKey]*utilipvs.VirtualServer
-	Destinations map[serviceKey][]*utilipvs.RealServer
+	Services     map[ServiceKey]*utilipvs.VirtualServer
+	Destinations map[ServiceKey][]*utilipvs.RealServer
 }
 
-type serviceKey struct {
+type ServiceKey struct {
```
Comment (andrewsykim) [inline pkg/util/ipvs/testing/fake.go]: Was easier to work with test fixtures by exporting this

**Comment by andrewsykim** [comment]
cc @lbernail

**Comment by lbernail** [comment]
Thanks a lot for this!
We could also add:
- an example with UDP (if we merge this after the udp PR: https://github.com/kubernetes/kubernetes/pull/77802)
- test for service deletion (which should put all associated RS in graceful termination)

We can do that later of course

**Comment by lbernail** [comment]
We probably should move `TestCleanLegacyRealServersExcludeCIDRs` to graceful_termination_test because it tests against a `GracefulTerminationManager`

**Comment by lbernail** [comment]
Let's merge this and add additional test later. This is already a big improvement
/lgtm

**Comment by m1093782566** [comment]
OK

/approve

**Comment by andrewsykim** [comment]
@lbernail will follow-up with some of your suggestions soon :)

**Pull request merged**